File: entities/12340/spell-item-enchantment.js
const t = require("../../types");

module.exports = {
  ID: t.int32le,
  Charges: t.int32le,
  Effect_1: t.int32le,
  Effect_2: t.int32le,
  Effect_3: t.int32le,
  EffectPointsMin_1: t.int32le,
  EffectPointsMin_2: t.int32le,
  EffectPointsMin_3: t.int32le,
  EffectPointsMax_1: t.int32le,
  EffectPointsMax_2: t.int32le,
  EffectPointsMax_3: t.int32le,
  EffectArg_1: t.int32le,
  EffectArg_2: t.int32le,
  EffectArg_3: t.int32le,
  Name_Lang_enUS: t.string,
  Name_Lang_enGB: t.string,
  Name_Lang_koKR: t.string,
  Name_Lang_frFR: t.string,
  Name_Lang_zhCN: t.string,
  Name_Lang_enCN: t.string,
  Name_Lang_deDE: t.string,
  Name_Lang_enTW: t.string,
  Name_Lang_zhTW: t.string,
  Name_Lang_esES: t.string,
  Name_Lang_esMX: t.string,
  Name_Lang_ruRU: t.string,
  Name_Lang_ptPT: t.string,
  Name_Lang_ptBR: t.string,
  Name_Lang_itIT: t.string,
  Name_Lang_Unk: t.string,
  Name_Lang_Mask: t.int32le,
  ItemVisual: t.int32le,
  Flags: t.int32le,
  Src_ItemID: t.int32le,
  Condition_Id: t.int32le,
  RequiredSkillID: t.int32le,
  RequiredSkillRank: t.int32le,
  MinLevel: t.int32le,
};

File: lib/src/definition/spell/spell_item_enchantment.dart
import 'package:warcrafty/src/core/field_definition.dart';
import 'package:warcrafty/src/core/structure_definition.dart';
import 'package:warcrafty/src/core/field_format.dart';
import 'package:warcrafty/src/core/locale_fields.dart';

/// SpellItemEnchantment 结构定义
///
/// 基于 WoWDBDefs 定义，版本 3.3.5.12340
final spellItemEnchantment = DbcStructureDefinition(
  name: 'SpellItemEnchantment',
  format: 'niiiiiiiiiiiiisssssssssssssssssiiiiiii',
  fields: [
    const FieldDefinition(index: 0, name: 'ID', description: 'ID', format: DbcFieldFormat.indexField),
    const FieldDefinition(index: 1, name: 'Charges', description: 'Charges', format: DbcFieldFormat.intType),
    const FieldDefinition(index: 2, name: 'Effect0', description: 'Effect 0', format: DbcFieldFormat.intType),
    const FieldDefinition(index: 3, name: 'Effect1', description: 'Effect 1', format: DbcFieldFormat.intType),
    const FieldDefinition(index: 4, name: 'Effect2', description: 'Effect 2', format: DbcFieldFormat.intType),
    const FieldDefinition(index: 5, name: 'EffectPointsMin0', description: 'EffectPointsMin 0', format: DbcFieldFormat.intType),
    const FieldDefinition(index: 6, name: 'EffectPointsMin1', description: 'EffectPointsMin 1', format: DbcFieldFormat.intType),
    const FieldDefinition(index: 7, name: 'EffectPointsMin2', description: 'EffectPointsMin 2', format: DbcFieldFormat.intType),
    const FieldDefinition(index: 8, name: 'EffectPointsMax0', description: 'EffectPointsMax 0', format: DbcFieldFormat.intType),
    const FieldDefinition(index: 9, name: 'EffectPointsMax1', description: 'EffectPointsMax 1', format: DbcFieldFormat.intType),
    const FieldDefinition(index: 10, name: 'EffectPointsMax2', description: 'EffectPointsMax 2', format: DbcFieldFormat.intType),
    const FieldDefinition(index: 11, name: 'EffectArg0', description: 'EffectArg 0', format: DbcFieldFormat.intType),
    const FieldDefinition(index: 12, name: 'EffectArg1', description: 'EffectArg 1', format: DbcFieldFormat.intType),
    const FieldDefinition(index: 13, name: 'EffectArg2', description: 'EffectArg 2', format: DbcFieldFormat.intType),
    ...createLocaleFieldsWithFlag(14, 'Name_lang', 'Name_lang'),
    const FieldDefinition(index: 31, name: 'ItemVisual', description: 'ItemVisual', format: DbcFieldFormat.intType),
    const FieldDefinition(index: 32, name: 'Flags', description: 'Flags', format: DbcFieldFormat.intType),
    const FieldDefinition(index: 33, name: 'Src_itemID', description: 'Src_itemID', format: DbcFieldFormat.intType),
    const FieldDefinition(index: 34, name: 'Condition_ID', description: 'Condition_ID', format: DbcFieldFormat.intType),
    const FieldDefinition(index: 35, name: 'RequiredSkillID', description: 'RequiredSkillID', format: DbcFieldFormat.intType),
    const FieldDefinition(index: 36, name: 'RequiredSkillRank', description: 'RequiredSkillRank', format: DbcFieldFormat.intType),
    const FieldDefinition(index: 37, name: 'MinLevel', description: 'MinLevel', format: DbcFieldFormat.intType),
  ],
);
